File: 201-bitwise-and-of-numbers-range/201-bitwise-and-of-numbers-range.cpp
class Solution {
public:
    int rangeBitwiseAnd(int left, int right) {
        int shiftCount = 0;
        
        while(left != right && left != 0) {
            left = left >> 1;
            right = right >> 1;
            ++shiftCount;
        }
        return left << shiftCount;
    }
};

File: 201-bitwise-and-of-numbers-range/201-bitwise-and-of-numbers-range.py
class Solution:
    def rangeBitwiseAnd(self, left: int, right: int) -> int:
        shiftCount = 0
        
        while left != right and left != 0:
            left = left >> 1
            right = right >> 1
            shiftCount += 1
        
        return left << shiftCount